src/audio/sr_audio/sr_audio.c

```diff
@@ -0,0 +1,720 @@
+// Copyright(c) 2021 Sound Research Corporation. All rights reserved.
+#include <sof/audio/component.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/sr_audio/sr_audio.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+//#define _ADD_TEST_CLICK_AT48
+
+static const struct comp_driver comp_sraudio;
+
+// {9C5B18CC-12A5-4904-A84A-7E1FE86E0927}
+DECLARE_SOF_RT_UUID("sraudio", sraudio_uuid, 0x9c5b18cc, 0x12a5, 0x4904, 0xa8, 0x4a, 0x7e, 0x1f, 0xe8, 0x6e, 0x9, 0x27);
+
+DECLARE_TR_CTX(sraudio_tr, SOF_UUID(sraudio_uuid), LOG_LEVEL_INFO);
+
+#define PARAM_CONTEXT_SIZE   55000 // iSST
+
+#define USE_STATIC_CTX
+
+typedef struct SR_HIFI2_Context_t {
+#if !defined(_MSC_VER) && !defined(USE_STATIC_CTX)
+  int32_t ctxBuffer[PARAM_CONTEXT_SIZE / sizeof(int32_t) + 1];
+#else
+  void* ctxBuffer;
+#endif
+  int32_t configParamID;
+  int32_t writeParamCount;
+  int32_t writeParamID;
+  int32_t writeParamDataSize;
+  int32_t writeParamDataOffset;
+  int32_t writeParamDataOffsetEnd;
+  int32_t readParamCount;
+  int32_t readParamID;
+  int32_t readParamDataSize;
+  int32_t readParamDataOffset;
+  int32_t readParamDataOffsetEnd;
+} SR_HIFI2_Context_t;
+
+// Initializes/resets context
+void SR_HIFI2_Initialize(SR_HIFI2_Context_t* libCtx);
+
+// Clears internal buffers. For flags, see SR_HIFI2_CLEAR_...
+void SR_HIFI2_Clear(SR_HIFI2_Context_t* libCtx, int flags);
+
+// Clears internal buffers asynchronously (inside the processing), for flags see SR_HIFI2_CLEAR_...
+void SR_HIFI2_AsyncClear(SR_HIFI2_Context_t* libCtx, int flags);
+
+// The below two functions allow sending/receiving parameter data in fragments.
+//
+// paramId: parameter id
+// paramData: points to parameter data (or to current data fragment, if data is fragmented)
+// dataSize: overall data size
+// dataFragmentSize: size of current fragment
+// isFirstFragment: 1 if paramData points to first fragment, 0 otherwise
+//
+// If non-fragmented data (aka single fragment) is passed, just call functions as follows below:
+// SR_HIFI2_SetParameter(context, paramId, paramData, dataSize, dataSize, 1);
+//   or
+// SR_HIFI2_GetParameter(context, paramId, paramData, dataSize, dataSize, 1);
+//
+void SR_HIFI2_SetParameter(SR_HIFI2_Context_t* libCtx, int paramId, const uint8_t* paramData, int dataSize, int dataFragmentSize, int isFirstFragment);
+void SR_HIFI2_GetParameter(SR_HIFI2_Context_t* libCtx, int paramId, uint8_t* paramData, int dataSize, int dataFragmentSize, int isFirstFragment);
```

Comment:
these line are quite long - 120 chars is limit.

Reply:
ok